functions/src/handlers.ts

```diff
@@ -15,20 +15,40 @@
  */
 
 import * as cors from 'cors';
-import { DecodedIdToken } from 'firebase-admin/auth';
-import { https, Response } from 'firebase-functions';
-import { getDecodedIdToken } from './common/auth';
-import { INTERNAL_SERVER_ERROR, UNAUTHORIZED } from 'http-status-codes';
+import {DecodedIdToken} from 'firebase-admin/auth';
+import {https, Response} from 'firebase-functions';
+import {getDecodedIdToken} from 'src/common/auth';
+import {INTERNAL_SERVER_ERROR, UNAUTHORIZED} from 'http-status-codes';
 import * as cookieParser from 'cookie-parser';
 
-const corsOptions = { origin: true };
+const corsOptions = {origin: true};
 const corsMiddleware = cors(corsOptions);
 
+/** Token to be used when running on local emulator for debugging. */
+class EmulatorIdToken implements DecodedIdToken {
+  aud = '';
+  auth_time = 0;
+  exp = 0;
+  firebase = {identities: {}, sign_in_provider: ''};
+  iat = 0;
+  iss = '';
+  sub = '';
+  uid = 'emu';
+}
+
 /**
  * Checks for and extracts the current user's details, passing them to the provided function if found.
  * Sends an UNAUTHORIZED HTTP response code if not present or invalid.
  */
-async function requireIdToken(req: https.Request, res: Response, next: (decodedIdToken: DecodedIdToken) => Promise<any>): Promise<any> {
+async function requireIdToken(
+  req: https.Request,
+  res: Response,
+  next: (decodedIdToken: DecodedIdToken) => Promise<any>
+): Promise<any> {
+  if (process.env.FUNCTIONS_EMULATOR === 'true') {
```

Comment:
@os-micmec Sorry for the noise, the new code is here and in EmulatorIdToken above.